l4_to_tda_spec_vap_e2e failing

**Describe the bug**
```
FAILED tests/e2e_tests/l4_to_tda_spec_vap_e2e.py::test_l4_to_tda_spec_vap - ValueError: Core throughput correction must be applied before convert_spec_to_flux for coronagraphic images (m...
```

Stack trace: 
```
_____________________________________________ test_l4_to_tda_spec_vap _____________________________________________

e2edata_path = '/home/jwang/Desktop/E2E_Test_Data/', e2eoutput_path = 'tests/e2e_tests/'

    @pytest.mark.e2e
    def test_l4_to_tda_spec_vap(e2edata_path, e2eoutput_path):
        """Pytest entry point for the spectroscopy VAP workflow.
    
        Args:
            e2edata_path (str): Fixture-provided path to E2E input data products.
            e2eoutput_path (str): Fixture-provided path where test outputs should be stored.
        """
        global logger
    
        output_top_level = os.path.join(e2eoutput_path, 'l4_to_tda_spec_vap')
        if not os.path.exists(output_top_level):
            os.makedirs(output_top_level)
        for f in os.listdir(output_top_level):
            file_path = os.path.join(output_top_level, f)
            if os.path.isfile(file_path):
                os.remove(file_path)
    
        log_file = os.path.join(output_top_level, 'l4_to_tda_spec_vap.log')
    
        logger = logging.getLogger('l4_to_tda_spec_vap')
        logger.setLevel(logging.INFO)
        logger.handlers.clear()
    
        file_handler = logging.FileHandler(log_file)
        file_handler.setLevel(logging.INFO)
        console_handler = logging.StreamHandler()
        console_handler.setLevel(logging.INFO)
        formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
        file_handler.setFormatter(formatter)
        console_handler.setFormatter(formatter)
        logger.addHandler(file_handler)
        logger.addHandler(console_handler)
    
        logger.info('=' * 80)
        logger.info('SPECTROSCOPY L4->TDA VAP TEST')
        logger.info('=' * 80)
        logger.info("")
    
>       run_spec_l4_to_tda_vap_test(e2edata_path, output_top_level)

tests/e2e_tests/l4_to_tda_spec_vap_e2e.py:514: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/e2e_tests/l4_to_tda_spec_vap_e2e.py:233: in run_spec_l4_to_tda_vap_test
    calibrated_img = l4_to_tda.convert_spec_to_flux(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input_dataset = <corgidrp.data.Dataset object at 0x77a8900975c0>
fluxcal = <corgidrp.data.FluxcalFactor object at 0x77a89008a660>

    def convert_spec_to_flux(input_dataset, fluxcal):
        """
        Flux calibrate 1-D spectroscopy spectra stored in the L4 SPEC extension.
        The function propagates calibration uncertainties. It applies the spectral flux calibration,
        if not available it can apply the fluxcal factor and color correction.
        Requires the input dataset to have already been core-throughput
        (ie SPEC header contains CTCOR=True) and
        slit transmission corrected (ie SPEC header contains SLITCOR=True).
    
        Args:
            input_dataset (corgidrp.data.Dataset): L4 dataset containing SPEC,
                SPEC_ERR, SPEC_DQ, SPEC_WAVE, and SPEC_WAVE_ERR extensions.
            fluxcal (corgidrp.data.SpecFluxCal or corgidrp.data.FluxcalFactor): wavelength dependent flux calibration
                or absolute flux calibration product used to convert the spectrum to absolute flux units erg/(s*cm^2*Å).
    
        Returns:
            corgidrp.data.Dataset: copy of the input dataset with the
            SPEC/SPEC_ERR data converted to erg/(s*cm^2*Å) and
            headers/history updated.
    
        """
        specflux = False
        #check that the correct flux calibration file is used
        if isinstance(fluxcal, SpecFluxCal):
           specflux = True
        else:
            if not isinstance(fluxcal, FluxcalFactor):
                raise TypeError("fluxcal must be a corgidrp.data.FluxcalFactor or SpecFluxCal instance.")
    
        spec_dataset = input_dataset.copy()
    
        for frame in spec_dataset:
            if 'SPEC' not in frame.hdu_list:
                raise ValueError("Input dataset does not contain a 'SPEC' extension.")
    
            is_coron = frame.pri_hdr.get('VISTYPE') not in ['CGIVST_CAL_SPEC_TGTREF','CGIVST_CAL_TGTREF_PHOT'] # using VISTYPE keyword to check if the image is coronagraphic
            if is_coron:
                if not frame.hdu_list['SPEC'].header.get('CTCOR', False):
>                   raise ValueError("Core throughput correction must be applied before convert_spec_to_flux for coronagraphic images (missing CTCOR flag).")
E                   ValueError: Core throughput correction must be applied before convert_spec_to_flux for coronagraphic images (missing CTCOR flag).

corgidrp/l4_to_tda.py:183: ValueError
```

**To Reproduce**
Run l4_to_tda_spec_vap_e2e.py on develop branch.

**Expected behavior**
I think we just need to modify the VISTYPES probably to reflect which are coron and which are non-coron. 

**corgidrp version:**
develop

--- a/tests/e2e_tests/l4_to_tda_spec_vap_e2e.py
+++ b/tests/e2e_tests/l4_to_tda_spec_vap_e2e.py
@@ -174,6 +174,14 @@ def run_spec_l4_to_tda_vap_test(e2edata_path, e2eoutput_path):
                     f"    SPEC BUNIT before flux calibration: {spec_bunit_input}. PASS."
                 )
 
+            # Updating VISTYPE since they are needed for CTCOR check in convert_spec_to_flux()
+            if is_coron:
+                img.pri_hdr['VISTYPE'] = 'CGIVST_TDD_OBS'
+                logger.info(f"VISTYPE is {img.pri_hdr['VISTYPE']}")
+            else:
+                img.pri_hdr['VISTYPE'] = 'CGIVST_CAL_SPEC_TGTREF'
+                logger.info(f"VISTYPE is {img.pri_hdr['VISTYPE']}")
+                
             # PSF-subtracted images: core throughput + CT_THRU grid + CTCOR 
             if is_coron:
                 try:

